src/ng/compile.js

```diff
@@ -296,6 +296,8 @@
  * * `$onInit` - Called on each controller after all the controllers on an element have been constructed and
  *   had their bindings initialized (and before the pre &amp; post linking functions for the directives on
  *   this element). This is a good place to put initialization code for your controller.
+ * * `$onDestroy` - Called when the controller's scope has been destroyed. This is a good place to put
+ * code for tearing down your controller and for example removing event listeners.
```

Comment:
We should also be explicit about when this is called wrt to child component/directives.